### PR TITLE
Adding arm64 image build support

### PR DESCRIPTION
**Pull request by quincycheng**
### What does this PR do?
- Adding linux/arm64 image build

### What ticket does this PR close?
nope

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: Dockerfile.arm64 [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+             $LOG_DIR \
+             $SSL_CERT_DIRECTORY/ca \
+             $SSL_CERT_DIRECTORY/cert \
+             /run/authn-local
+
+COPY Gemfile \
+     Gemfile.lock ./
+COPY gems/ gems/
+
+
```
Comment (orenbm) [inline Dockerfile.arm64:37]: remove empty line

File: Dockerfile.arm64 [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# Ensure few required GID0-owned folders to run as a random UID (OpenShift requirement)
+RUN mkdir -p $TMP_DIR \
+             $LOG_DIR \
+             $SSL_CERT_DIRECTORY/ca \
+             $SSL_CERT_DIRECTORY/cert \
+             /run/authn-local
+
+COPY Gemfile \
+     Gemfile.lock ./
+COPY gems/ gems/
```
Comment (orenbm) [inline Dockerfile.arm64:35]: please add a line before this

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- arm64 image build support
+
+### Changed
+- remove usage of syscall() to support non-amd64 systems
```
Comment (orenbm) [inline CHANGELOG.md:13]: @quincycheng some comments on this:

1. Does this PR have an issue that it resolves? If not, please create one. In any case, please add a link to it to align with the standard (you can see examples below).
2. Please capitalized the first words of the sentence.

**Comment by orenbm** [comment]
@quincycheng can you please add an example of how the log lines looked before this change and how they look now? I would like to verify that we don't miss anything.

File: buildx.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (orenbm) [inline buildx.sh:1]: @quincycheng where is this script used? is it for dev purposes? can you add a usage guide?

File: buildx.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
+DOCKER_USER=$1
+
+docker buildx build -f Dockerfile.arm64 -t $DOCKER_USER/conjur:latest \
+  --build-arg DOCKER_USER="$DOCKER_USER" \
+  --platform linux/amd64,linux/arm64 .  --push
```
Comment (orenbm) [inline buildx.sh:7]: please change this to:
```
docker buildx build \
  -f Dockerfile.arm64 \
  -t $DOCKER_USER/conjur:latest \
  --build-arg DOCKER_USER="$DOCKER_USER" \
  --platform linux/amd64,linux/arm64 . \
  --push
```

to enhance readability.

**Comment by izgeri** [comment]
Just noting here that this issue is connected to cyberark/conjur-base-image#45, and I've asked on that PR that we file an issue to describe the motivation for this change so that we can better understand how to integrate this into the two projects (Conjur and Conjur Base Image).

**Comment by quincycheng** [comment]
> @quincycheng can you please add an example of how the log lines looked before this change and how they look now? I would like to verify that we don't miss anything.

No log can be generated before this change.   
Before: 
After: 
```
conjur_server | [origin=172.30.0.1] [request_id=cd5a2e73-4185-49be-873a-4761c89502ff] [tid=1609418859] Completed 200 OK in 22ms (Views: 18.4ms)
```

**Comment by quincycheng** [comment]
Issue https://github.com/cyberark/conjur/issues/1988 created

File: buildx.sh (hunk 4 above)
Comment (izgeri) [reply buildx.sh:1]: +1 on this comment - can we just add building this script to build.sh? what's the benefit of having a whole separate script?

File: CHANGELOG.md (hunk 3 above)
Comment (izgeri) [inline CHANGELOG.md:13]: I think we should drop this item from the changelog - it doesn't appear it has any user impact at all. please see our [changelog guidelines](https://github.com/cyberark/community/blob/master/Conjur/CONTRIBUTING.md#changelog-guidelines) to learn more.

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- arm64 image build support
```
Comment (izgeri) [inline CHANGELOG.md:10]:
```suggestion
- Ability to build an ARM64 Conjur image to enable running Conjur on ARM-based devices
   like Raspberry Pi and new Macs.
   [cyberark/conjur#1988](https://github.com/cyberark/conjur/issues/1988)
```

**Review by izgeri — CHANGES_REQUESTED** [review]
@quincycheng I think the main goal you have for adding this Dockerfile to the project is to enable people to build a Conjur image that supports ARM. To achieve this goal, I think it's important that we document in the repo that this functionality exists. Please add notes on building an ARM64 image (including required commands) in a subsection of the CONTRIBUTING guide. once this is there, we can use it to point people to / share in discourse.

Also, does merging this PR depend on first merging https://github.com/cyberark/conjur-base-image/pull/45?

**Pull request closed without merging**